paramiko/pipe.py

```diff
@@ -64,7 +63,13 @@ def set(self):
         if self._set or self._closed:
             return
         self._set = True
-        os.write(self._wfd, b"*")
+        try:
+            os.write(self._wfd, b"*")
+        except OSError as e:
+            if e.errno == 9 and self._closed:
+                # The pipe was closed, no need to do anything
+                return
+            raise e
```

Comment:
Do we not also need something analogous to protect the `os.read(self._rfd, 1)` call above in `clear()`?

(We might not; again, just raising the question.)

Reply:
From the `pipe`'s perspective, yes. And it would be wise to protect it. However, the way the pipe is used, this does not happen. The same thread
1. Opens,
2. clears and
3. closes
the pipe. So there is no race condition between `.clear()` and `.close().
A second thread sets it.

Reply:
Same request, please add a comment here explaining why a similar race catch is not (currently) required here.

Reply:
done